server/src/main/java/org/opensearch/cluster/service/TaskBatcher.java

```diff
@@ -136,6 +148,7 @@ private void onTimeoutInternal(List<? extends BatchedTask> tasks, TimeValue time
                     }
                 }
             }
+            taskBatcherListener.onProcessed(toRemove);
```

Comment:
should this call onTimeout in listener?

Reply:
We have kept only onProcessed (which will be invoked whenever task is marked as processed). Here in timeout case as well it is marking tasks as processed so calling it. We don't have onTimeout in listener.